### PR TITLE
Fix LDAP search bind behavior

### PR DESCRIPTION
**Pull request by gfrey**
My directory has an uid entry for users, that should be used for finding a proper DN for doing the actual bind in search mode, ie. the search returns something like 'cn=John Doe,ou=users,dc=example' for uid 'jdoe'. The old code did a simple bind with 'uid=jdoe,ou=users,dc=example' which didn't work. Now the returned DN is used for the simple bind.

### CODE REVIEW

**Comment by erikjohnston** [comment]
@matrixbot ok to test

**Comment by erikjohnston** [comment]
@mweinelt could you take a look?

**Comment by mweinelt** [comment]
@erikjohnston probably tonight.

File: synapse/util/ldap_auth_provider.py [hunk 1]
```diff
@@ -219,9 +219,8 @@ class _LdapConfig(object):
 
         return ldap_config
 
-    def _ldap_simple_bind(self, server, localpart, password):
-        """ Attempt a simple bind with the credentials
-            given by the user against the LDAP server.
+    def _ldap_internal_bind(self, server, bind_dn, password):
```
Comment (mweinelt) [inline synapse/util/ldap_auth_provider.py]: Prefixing with `_` and calling it `internal` looks like a duplication, why not call it `_ldap_bind`?

Comment (gfrey) [inline synapse/util/ldap_auth_provider.py]: Good one! Done.

**Comment by gfrey** [comment]
I use this with my private synapse instance using the search bind method. I did not test simple bind. Maybe I find some time this evening.

**Comment by erikjohnston** [comment]
(The failing tests look to be unrelated and probably due to bug fixes on develop that haven't been merged in to this branch)

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Comment by erikjohnston** [comment]
Hi, sorry this got dropped on the floor, but we've now moved the ldap code to a different repository: https://github.com/matrix-org/matrix-synapse-ldap3, I'm not sure if this has been implemented there or not. Again, sorry about this not getting merged before porting.

**Comment by kyrias** [comment]
For the record, I got confirmation in #matrix-dev that the separated one works [here](https://matrix.to/#/!XqBunHwQIXUiqCaoxq:matrix.org/$148110723933KEYSm:matrix.molinkx.com).

**Pull request closed without merging**